### PR TITLE
Add Twirp Design Principles to the Contributing Guidelines and archive protocol v6

### PR DESCRIPTION
**Pull request by marioizquierdo**
Add a list of Twirp Principles to the CONTRIBUTING document, to clarify what are the priorities for the project. Include examples of contributions that are welcome, and examples of contributions that will be difficult. Also update the instructions to run tests and organize the document a little better.

Archive the draft proposition for Twirp v6. Keep a document of the intended changes, to acknowledge those who may have waited for this to happen. Make it clear that v6 is not going to happen.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,38 +1,64 @@
...
+ * Prefer generated code over shared libraries between services and clients, so it is easier to implement changes without breaking compatibility.
 
-## Setup Requirements ##
+Contributions that are welcome:
 
-You will need git, Go 1.9+, and Python 2.7 installed and on your system's path.
-Install them however you feel.
+ * Security updates.
+ * Performance improvements.
+ * Supporting new versions of Golang and Protobug.
```
Comment (ofpiyush) [inline CONTRIBUTING.md:24]:
```suggestion
 * Supporting new versions of Golang and Protobuf.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,38 +1,64 @@
...
+ * Documentation.
+ * Making Twirp easier to integrate with other useful tools.
 
-## Developer Loop ##
+In the other hand, contributions that contradict the following priorities will be more difficult:
```
Comment (ofpiyush) [inline CONTRIBUTING.md:28]:
```suggestion
On the other hand, contributions that contradict the following priorities will face more resistance:
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,38 +1,64 @@
...
+## Contributing Code with Pull Requests
+
+Twirp uses github pull requests. Fork, hack away at your changes and submit. Most pull requests will go through a few iterations before they get merged. Different contributors will sometimes have different opinions, and often patches will need to be revised before they can get merged.
+
+### Requirements
 
-Most tests of the Go server are in `internal/twirptest/service_test.go`. Tests
-of cross-language clients are in the [clientcompat](./clientcompat) directory.
+ * Twirp officially supports the last 3 releases of Go.
+ * The Python implementation uses Python 2.7.
```
Comment (ofpiyush) [inline CONTRIBUTING.md:46]: Python has stopped official support for `2.7`, maybe remove the requirement?

https://github.com/python/devguide/pull/344

Comment (marioizquierdo) [reply CONTRIBUTING.md:46]: That is right. However this should be discussed in a different PR. The Python implementation has a few things that should be discussed all together. This PR adds some edits to the Contributing Guidelines, but with the focus on adding the Twirp Principles.

Comment (marioizquierdo) [reply CONTRIBUTING.md]: We edited this point with: "As such, it is not suitable for production use in its current form.". Thanks for the reference.

**Pull request merged**